### DISCUSSION TITLE
Is there a way to Save a copy of an item through API?

### DISCUSSION THREAD

**Discussion by adanielyan** [directus/directus#14928] [Archive]
I searched through documentation but didn't find how a "Saved as copy" can be done through API. Is this feature only available in the detectus app but not API?

**Comment by br41nslug** [comment]
The "save a copy" function is fully on the app/frontend side but you can mimic what the app is doing, fetching the record and posting it back without primary key

**Reply by adanielyan** [reply]
> The "save a copy" function is fully on the app/frontend side but you can mimic what the app is doing, fetching the record and posting it back without primary key

Thank you for a suggestion. In my case I have many complex related items that should be duplicated too when the main item is saved as a copy, using the Duplication configuration on my data models. Do you know if there is a directus service that I can reuse in a custom extension to have the same functionality without reinventing the wheel?

**Reply by rijkvanzanten** [reply]
The duplication configuration option stores it in the same format as the `fields` parameter, so you should be able to read the configuration from directus_collections, and then directly use that value inside of the `fields` parameter of the items service 👍🏻
